wled00/FX_fcn.cpp

```diff
@@ -1079,13 +1170,15 @@ void WS2812FX::service() {
   _segment_index = 0;
   Segment::handleRandomPalette(); // move it into for loop when each segment has individual random palette
   for (segment &seg : _segments) {
+    if (!seg.isActive()) continue;
```

Comment:
Continuing before `seg.resetIfRequired()` causes segment data to not be freed if the segment is deleted. Could this be the cause of the leak you observed?